numaprom/watcher.py

```diff
@@ -0,0 +1,77 @@
+import os
+import time
+
+from omegaconf import OmegaConf
+from watchdog.observers import Observer
+from numalogic.config import NumalogicConf
+from watchdog.events import FileSystemEventHandler
+
+from numaprom._constants import CONFIG_DIR, DEFAULT_CONFIG_DIR
+from numaprom import NumapromConf, get_logger
+
+_LOGGER = get_logger(__name__)
+
+config = {}
```

Comment:
NIT:

```suggestion
class Configs(dict):
....
```

and attach load_configs and update_configs to that?